File: android/core/model/src/main/java/com/bolsaterminal/core/model/Screener.kt
package com.bolsaterminal.core.model

import kotlinx.serialization.Serializable

@Serializable
data class SearchResult(
    val symbol: String,
    val shortname: String,
    val longname: String? = null,
    val exchange: String,
    val quoteType: String,
)

@Serializable
data class ScreenerItem(
    val symbol: String,
    val shortName: String,
    val exchange: String,
    val sector: String? = null,
    val quoteType: String,
    val price: Double,
    val change: Double,
    val changePercent: Double,
    val volume: Double,
    val marketCap: Double? = null,
    val rsi: Double,
    val macdHistogram: Double,
    val bbPercent: Double,
    val signal: SignalType,
    val score: Double,
    val sparkline: List<Double>,
)

@Serializable
data class MarketIndex(
    val symbol: String,
    val name: String,
    val price: Double,
    val change: Double,
    val changePercent: Double,
    val sparkline: List<Double>,
    val marketId: String,
)

File: macos/BolsaTerminal/Core/Models/Screener.swift
import Foundation

struct SearchResult: Codable, Hashable, Identifiable {
    var id: String { symbol }
    let symbol: String
    let shortname: String
    let longname: String?
    let exchange: String
    let quoteType: String
}

struct ScreenerItem: Codable, Hashable, Identifiable {
    var id: String { symbol }
    let symbol: String
    let shortName: String
    let exchange: String
    let sector: String?
    let quoteType: String
    let price: Double
    let change: Double
    let changePercent: Double
    let volume: Double
    let marketCap: Double?
    let rsi: Double
    let macdHistogram: Double
    let bbPercent: Double
    let signal: SignalType
    let score: Double
    let sparkline: [Double]
}

struct MarketIndex: Codable, Hashable, Identifiable {
    var id: String { symbol }
    let symbol: String
    let name: String
    let price: Double
    let change: Double
    let changePercent: Double
    let sparkline: [Double]
    let marketId: String
}
